connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java

```diff
@@ -262,127 +254,96 @@ public void testFullExpandConnectors() {
-                .andReturn(new RestClient.HttpResponse<>(201, new HashMap<>(), new ConnectorInfo(CONNECTOR_NAME, CONNECTOR_CONFIG, CONNECTOR_TASK_NAMES,
-                    ConnectorType.SOURCE)));
+        final ArgumentCaptor<Callback<Herder.Created<ConnectorInfo>>> cb = ArgumentCaptor.forClass(Callback.class);
+        expectAndCallbackNotLeaderException(cb).when(herder)
+            .putConnectorConfig(eq(CONNECTOR_NAME), eq(body.config()), eq(false), cb.capture());
 
-        PowerMock.replayAll();
+        // Should forward request
+        restClientStatic.when(() ->
+            RestClient.httpRequest(eq(LEADER_URL + "connectors?forward=false"), eq("POST"), isNull(), eq(body), any(), any(WorkerConfig.class))
+        ).thenReturn(new RestClient.HttpResponse<>(201, new HashMap<>(), new ConnectorInfo(CONNECTOR_NAME, CONNECTOR_CONFIG, CONNECTOR_TASK_NAMES, ConnectorType.SOURCE)));
 
         connectorsResource.createConnector(FORWARD, NULL_HEADERS, body);
-
-        PowerMock.verifyAll();
-
-
     }
 
     @Test
     public void testCreateConnectorWithHeaderAuthorization() throws Throwable {
         CreateConnectorRequest body = new CreateConnectorRequest(CONNECTOR_NAME, Collections.singletonMap(ConnectorConfig.NAME_CONFIG, CONNECTOR_NAME));
-        final Capture<Callback<Herder.Created<ConnectorInfo>>> cb = Capture.newInstance();
-        HttpHeaders httpHeaders = EasyMock.mock(HttpHeaders.class);
-        EasyMock.expect(httpHeaders.getHeaderString("Authorization")).andReturn("Basic YWxhZGRpbjpvcGVuc2VzYW1l").times(1);
-        EasyMock.replay(httpHeaders);
-        herder.putConnectorConfig(EasyMock.eq(CONNECTOR_NAME), EasyMock.eq(body.config()), EasyMock.eq(false), EasyMock.capture(cb));
-        expectAndCallbackNotLeaderException(cb);
+        final ArgumentCaptor<Callback<Herder.Created<ConnectorInfo>>> cb = ArgumentCaptor.forClass(Callback.class);
+        HttpHeaders httpHeaders = mock(HttpHeaders.class);
+        expectAndCallbackNotLeaderException(cb)
+            .when(herder).putConnectorConfig(eq(CONNECTOR_NAME), eq(body.config()), eq(false), cb.capture());
 
-        EasyMock.expect(RestClient.httpRequest(EasyMock.eq(LEADER_URL + "connectors?forward=false"),
-            EasyMock.eq("POST"), EasyMock.eq(httpHeaders), EasyMock.anyObject(), EasyMock.anyObject(), EasyMock.anyObject(WorkerConfig.class)))
-                .andReturn(new RestClient.HttpResponse<>(202, new HashMap<>(), null));
-
-        PowerMock.replayAll();
+        restClientStatic.when(() ->
+            RestClient.httpRequest(eq(LEADER_URL + "connectors?forward=false"), eq("POST"), eq(httpHeaders), any(), any(), any(WorkerConfig.class))
+        ).thenReturn(new RestClient.HttpResponse<>(202, new HashMap<>(), null));
 
         connectorsResource.createConnector(FORWARD, httpHeaders, body);
-
-        PowerMock.verifyAll();
     }
 
 
 
     @Test
     public void testCreateConnectorWithoutHeaderAuthorization() throws Throwable {
```

Comment:
Do we still need this test? It looks like it's identical to `testCreateConnectorWithHeaderAuthorization`; maybe we could consolidate the two into something like `testCreateConnectorWithForwardedHeaders`?

Reply:
I will do this when you are happy with https://github.com/apache/kafka/pull/12725#discussion_r991457894

Reply:
Okay so I had a look into this because I ended up getting confused with what happened. Originally the test was using specific Authorization headers, i.e.

```java
when(httpHeaders.getHeaderString("Authorization")).thenReturn("Basic YWxhZGRpbjpvcGVuc2VzYW1l");
```

and

```java
when(httpHeaders.getHeaderString("Authorization")).thenReturn(null);
```

respectively. I then enabled `@RunWith(MockitoJUnitRunner.StrictStubs.class)` at which point Mockito reported that these stubs weren't being used which ended up making the two tests identical. So the question then is, is it useful for the specific values of the `"Authorization"` header or just that the `HttpHeaders` object is correctly being passed and returned?

If you don't care about testing the the actual value of the `"Authorization"` header then I can just remove the identical test and rename the other one to `testCreateConnectorWithForwardedHeaders` as you stated otherwise I would need to improve the tests.

Reply:
I think we can just consolidate the test cases. The coverage we need here is to ensure that the headers that were given to `connectorsResource::createConnector` are propagated to `RestClient::httpRequest`. The actual content of those headers doesn't matter in this test suite.

Probably worth renaming to something like `testCreateConnectorWithHeaders` since authorization no longer really has anything to do with it.

Reply:
Fixed in the `Consolidate header authorization test into testCreateConnectorWithHeaders` commit